.circleci/config.yml

```diff
@@ -0,0 +1,95 @@
+version: 2
+
+
+defaults:
+
+  workspace_root: &workspace_root
+    ~/build
+
+  container: &default_container
+    working_directory: *workspace_root
+    docker:
+      - image: circleci/openjdk:8-jdk
+
+
+jobs:
+
+  build_connectors:
+    <<: *default_container
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - connectors-dependencies-{{ checksum "build.gradle" }}
+            - connectors-dependencies
+          paths:
+            - ~/.gradle/caches/
+            - ~/.gradle/wrapper/
+      - run:
+          name: Test and build connectors project
+          command: ./gradlew -p ./connectors clean build
+      - save_cache:
+          key: connectors-dependencies-{{ checksum "build.gradle" }}
+          paths:
+            - ~/.gradle/caches/
+            - ~/.gradle/wrapper/
+
+
+  build_streams-api:
+    <<: *default_container
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - api-dependencies-{{ checksum "build.gradle" }}
+            - api-dependencies
+          paths:
+            - ~/.gradle/caches/
+            - ~/.gradle/wrapper/
+      - run:
+          name: Test and build streams-api project
+          command: ./gradlew -p ./client-api clean build
+      - save_cache:
+          key: api-dependencies-{{ checksum "build.gradle" }}
+          paths:
+            - ~/.gradle/caches/
+            - ~/.gradle/wrapper/
+
+
+  deploy_connectors_image:
+     <<: *default_container
+     steps:
+       - checkout
+       - setup_remote_docker:
+           version: 17.05.0-ce
```

Comment:
https://circleci.com/docs/2.0/building-docker-images/#docker-version says that default is 17.03.0